### PR TITLE
Exponential backoffs in Sparrow.H2Worker

### PR DESCRIPTION
**Pull request by zofpolkowska**

### CODE REVIEW

File: lib/sparrow/h2_worker.ex [hunk 1]
```diff
@@ -414,6 +437,39 @@ defmodule Sparrow.H2Worker do
...
+
+  defp start_conn_backoff(config, {:ok, delay}, delay_stream) do
+    :timer.sleep(delay)
+
+    case start_conn(config) do
+      {:ok, state} ->
+        {:ok, state}
+
+      {:error, reason} ->
+        new_delay = Enum.fetch(delay_stream, 0)
```
Comment (rslota) [inline lib/sparrow/h2_worker.ex]: We probably still want to log something in this case. Maybe not an `error` but warning, or even info about failed attempt and retry would be nice to hav.e

File: lib/sparrow/h2_worker/config.ex [hunk 2]
```diff
@@ -54,15 +63,21 @@ defmodule Sparrow.H2Worker.Config do
         authentication,
         tls_options \\ [],
         ping_interval \\ 5_000,
-        reconnect_attempts \\ 3
+        reconnect_attempts \\ 3,
+        backoff_base \\ 2,
+        backoff_initial_delay \\ 100,
```
Comment (rslota) [inline lib/sparrow/h2_worker/config.ex]: Let's change the input here to a map - this way, optional values can be truly optional and can be defined in one place (a default map that we can merge input into).

File: test/h2_worker_test.exs [hunk 3]
```diff
@@ -489,8 +491,9 @@ defmodule Sparrow.H2WorkerTest do
             context[:auth]
           )
 
-        {:ok, pid} = GenServer.start_link(Sparrow.H2Worker, config)
-        state = :sys.get_state(pid)
+        worker_pid = start_supervised!(Tools.h2_worker_spec(config))
+        :timer.sleep(100)
+        state = :sys.get_state(worker_pid)
```
Comment (rslota) [inline test/h2_worker_test.exs]: Do we use that state? Please make sure all compile warnings that are generated here are fixed.

File: test/h2_worker_test.exs [hunk 4]
```diff
@@ -489,8 +491,9 @@ defmodule Sparrow.H2WorkerTest do
             context[:auth]
           )
 
-        {:ok, pid} = GenServer.start_link(Sparrow.H2Worker, config)
-        state = :sys.get_state(pid)
+        worker_pid = start_supervised!(Tools.h2_worker_spec(config))
+        :timer.sleep(100)
```
Comment (rslota) [inline test/h2_worker_test.exs]: Why do we sleep here if we block with the next call until gen server starts?

File: test/h2_worker_test.exs (hunk 3 above)
Comment (zofpolkowska) [reply test/h2_worker_test.exs]: Yes, we do use it

**Pull request merged**